### DISCUSSION TITLE
build warnings

### DISCUSSION THREAD

**Discussion by AndrewSav** [rspamd/rspamd#4353] [General]
We are getting [pages and pages](https://gist.github.com/AndrewSav/8efbb84ef0936bc7a5620723e86efe90) of warnings when building rspamd 3.4 from sources. It did not happen with rspamd 3.2. This is how we build it: <https://github.com/mailserver2/debian-mail-overlay/pull/17>. Is it know why later versions started to produce that many warnings?

**Comment by vstakhov** [comment]
If you read the warnings details you might have a glue... In fact, `-Weffc++` flag is somehow useful for the own code but it is totally useless for the 3rd party code. I should probably disable it completely, as it just hides the real issues.

**Reply by AndrewSav** [reply]
@vstakhov  Thank you for your reply, it sounds like you saying that this is nothing to worry about and that you going to reduce the amount of warnings by disabling `-Weffc++` for 3rd party code. Did I get this right?

I'm wondering what happened between 3.2 and 3.4 to cause this change, was this flag added just recently?

**Reply by vstakhov** [reply]
> @vstakhov Thank you for your reply, it sounds like you saying that this is nothing to worry about and that you going to reduce the amount of warnings by disabling `-Weffc++` for 3rd party code. Did I get this right?
> 
> I'm wondering what happened between 3.2 and 3.4 to cause this change, was this flag added just recently?

https://github.com/rspamd/rspamd/commit/8eeeed0da6862294f02992bad4769c7ef628029f

**Reply by AndrewSav** [reply]
@vstakhov Cool, really appreciate your time, thank you so much

**Reply by vstakhov** [reply]
I think it is better to disable this warning completely probably except asan builds.
